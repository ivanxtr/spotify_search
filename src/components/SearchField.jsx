import React from 'react'

function SearchField({ search, submitSearch, searchValue, setSearchValue }) {

  return(
    <>
    <div className="search-container" onKeyPress={(e) => submitSearch(e.key, searchValue)}>
      <input
          className="searchField"
          placeholder="Type your search"
          value={searchValue}
          type="search"
          onChange={(e) => setSearchValue(e.target.value)} 
      />
      <button onClick={() => search(searchValue)}>
        <span> Search on Spotify </span>
      </button>
    </div>
    </>
  )
}

export default SearchField
