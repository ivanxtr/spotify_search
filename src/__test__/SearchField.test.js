import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import SearchField from '../components/SearchField';

Enzyme.configure({ adapter: new Adapter() });

const dataFixture = []

describe('SearchField Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchField />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot render', () => {
    const component = renderer.create(<SearchField />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should display proper content', () => {
    const wrapper = mount(
      <SearchField
        showSearchField={true}
        data={dataFixture}
        getTrack={() => {}} 
      />);
    expect(wrapper.find('.search-container')).toHaveLength(1)
    expect(wrapper.find('.hide')).toHaveLength(0)
    expect(wrapper.find('.searchField')).toHaveLength(1)
    expect(wrapper.find('button')).toHaveLength(1)
    expect(wrapper.find('button span').text()).toBe(' Search on Spotify ')
  })
})
