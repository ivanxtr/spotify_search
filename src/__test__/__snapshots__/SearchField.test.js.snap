// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField Component snapshot render 1`] = `
<div
  className="search-container"
  onKeyPress={[Function]}
>
  <input
    className="searchField"
    onChange={[Function]}
    placeholder="Type your search"
    type="search"
  />
  <button
    onClick={[Function]}
  >
    <span>
       Search on Spotify 
    </span>
  </button>
</div>
`;
